cpg_pipes/targets.py

```diff
@@ -0,0 +1,640 @@
+            if (ds.active or not only_active)
+        ]
+
+    def get_dataset_by_name(
+        self, name: str, only_active: bool = True
+    ) -> Optional['Dataset']:
+        """
+        Get dataset by name.
+        Include only "active" datasets (unless only_active is False)
+        """
+        ds_by_name = {d.name: d for d in self.get_datasets(only_active)}
+        return ds_by_name.get(name)
+
+    def get_samples(self, only_active: bool = True) -> list['Sample']:
+        """
+        Gets a flat list of all samples from all datasets.
+        Include only "active" samples (unless only_active is False)
+        """
+        all_samples = []
+        for proj in self.get_datasets(only_active=only_active):
+            all_samples.extend(proj.get_samples(only_active))
+        return all_samples
+
+    def add_dataset(self, dataset: 'Dataset') -> 'Dataset':
+        """
+        Add existing dataset into the cohort.
+        """
+        if dataset.name in self._datasets_by_name:
+            logger.debug(f'Dataset {dataset.name} already exists in the cohort')
+            return dataset
+        self._datasets_by_name[dataset.name] = dataset
+        return dataset
+
+    def create_dataset(
+        self,
+        name: str,
+        namespace: Namespace | None = None,
+        storage_provider: StorageProvider | None = None,
+    ) -> 'Dataset':
+        """
+        Create a dataset and add it to the cohort.
+        """
+        namespace = namespace or self.analysis_dataset.namespace
+        # Normalising the dataset's name:
+        name = build_dataset_name(*parse_stack(name, namespace))
+        if name in self._datasets_by_name:
+            logger.debug(f'Dataset {name} already exists in the cohort')
+            return self._datasets_by_name[name]
+
+        if name == self.analysis_dataset.name:
+            ds = self.analysis_dataset
+        else:
+            ds = Dataset(
+                name=name,
+                namespace=namespace,
+                storage_provider=storage_provider or self.storage_provider,
+            )
+
+        self._datasets_by_name[ds.name] = ds
+        return ds
+
+    def get_job_attrs(self) -> dict[str, str]:
+        """
+        Attributes for Hail Batch job.
+        """
+        return {}
+
+    def get_job_prefix(self) -> str:
+        """
+        Prefix job names.
+        """
+        return ''
+
+
+def parse_stack(name: str, namespace: Namespace | None = None) -> tuple[str, Namespace]:
+    """
+    Input `name` can be either e.g. "seqr" or "seqr-test". The latter will be
+    resolved to stack="seqr" and is_test=True, unless `namespace` is provided
+    explicitly.
+
+    Returns the stack id and a corrected namespace.
+    """
+    namespace = namespace or Namespace.MAIN
+    if name.endswith('-test'):
+        stack = name[: -len('-test')]
+        if namespace == Namespace.MAIN:
+            namespace = Namespace.TEST
```

Comment:
Why not just always set to TEST?

```suggestion
        namespace = Namespace.TEST
```

Reply:
Thanks!